src/frontend/servers/y-provider/src/server.ts

```diff
@@ -1,18 +1,142 @@
+// eslint-disable-next-line import/order
+import './services/sentry';
 import { Server } from '@hocuspocus/server';
+import * as Sentry from '@sentry/node';
+import express, { Request, Response } from 'express';
+import expressWebsockets from 'express-ws';
 
-const port = Number(process.env.PORT || 4444);
+import { PORT } from './env';
+import { httpSecurity, wsSecurity } from './middlelayers';
+import { routes } from './routes';
+import { logger } from './utils';
 
-const server = Server.configure({
-  name: 'docs-y-provider',
-  port: port,
+export const hocuspocusServer = Server.configure({
+  name: 'docs-y-server',
   timeout: 30000,
-  debounce: 2000,
-  maxDebounce: 30000,
   quiet: true,
-});
+  onConnect({ requestHeaders, connection, documentName, requestParameters }) {
+    const roomParam = requestParameters.get('room');
+    const canEdit = requestHeaders['x-can-edit'] === 'True' ? true : false;
+
+    if (!canEdit) {
+      connection.readOnly = true;
+    }
+
+    logger(
+      'Connection established:',
+      documentName,
+      'userId:',
+      requestHeaders['x-user-id'],
+      'canEdit:',
+      canEdit,
+      'room:',
+      requestParameters.get('room'),
+    );
+
+    if (documentName !== roomParam) {
+      console.error(
+        'Invalid room name - Probable hacking attempt:',
+        documentName,
+        requestParameters.get('room'),
+        requestHeaders['x-user-id'],
+      );
+
+      return Promise.reject(new Error('Unauthorized'));
+    }
 
-server.listen().catch((error) => {
-  console.error('Failed to start the server:', error);
+    return Promise.resolve();
+  },
 });
 
-console.log('Websocket server running on port :', port);
+/**
+ * init the collaboration server.
+ *
+ * @param port - The port on which the server listens.
+ * @param serverSecret - The secret key for API authentication.
+ * @returns An object containing the Express app, Hocuspocus server, and HTTP server instance.
+ */
+export const initServer = () => {
+  const { app } = expressWebsockets(express());
+  app.use(express.json());
+
+  /**
+   * Route to handle WebSocket connections
+   */
+  app.ws(routes.WS, wsSecurity, (ws, req) => {
+    logger('Incoming Origin:', req.headers['origin']);
+
+    try {
+      hocuspocusServer.handleConnection(ws, req);
+    } catch (error) {
+      console.error('Failed to handle WebSocket connection:', error);
+      ws.close();
+    }
+  });
+
+  type ResetConnectionsRequestQuery = {
+    room?: string;
+  };
+
+  /**
+   * Route to reset connections in a room
+   */
+  app.post(
+    routes.RESET_CONNECTIONS,
```

Comment:
you need authentication on this view or it can be abused?

Reply:
This route is not for the clients, it is for the backend (signal), so we don't need / want the `collaboration_auth`.
The security is from the midlelayer [httpSecurity](https://github.com/numerique-gouv/impress/pull/472/files#diff-931f2e9c315f6fc95eae032ee4f167e486cab63966647873520cb12d39d56fdfR85), the "authentication" is here:
https://github.com/numerique-gouv/impress/pull/472/files#diff-4fa4cb104ce6aac1c7d731c6a4dbd6faaece1554a0052bfd6e4cf4f7a3f769acR28-R32

The keys have to match. We can increase the secu if you want. 
This endpoint does not gives you access to the document, you could reset a connection, that will trigger a reconnection on the client side.

Reply:
Ok understood. Then you can remove this ingress and access the service directly via its svc. Exposing an ingress is only to give access on internet.